service/history/queue/processor_base.go

```diff
@@ -304,6 +306,20 @@ func (p *processorBase) splitProcessingQueueCollection(
 	}
 }
 
+func (p *processorBase) emitProcessingQueueMetrics() {
+	numProcessingQueues := 0
+	maxProcessingQueueLevel := -1
```

Comment:
Maybe just init maxProcessingQueueLevel to 0? will scope.recordTimer(-1) causing issue?

Reply:
That's a good point. I will update. 

Theoretically, it can never happen as `processingQueueCollections` will never be empty; otherwise, it must be a code bug and we should panic or alert on that.

Reply:
Updated.